### ISSUE TITLE
Cargo audit failure to fetch database: failed to update references

### ISSUE THREAD

**Issue by MeGaGiGaGon** [rustsec/rustsec#1237]
I preformed the same modifications as done [in this issue](https://github.com/rustsec/rustsec/issues/1029#issuecomment-1814843858) to get the full trace.

```
C:\Users\user\Desktop\rust_projects\project>cargo audit
    Fetching advisory database from `https://github.com/RustSec/advisory-db.git`
error: couldn't fetch advisory database: git operation failed: failed to fetch repo: Failed to update references to their new position to match their remote locations
Caused by:
  -> The reflog could not be created or updated
  -> Could not open reflog file at "C:\\Users\\user\\.cargo\\advisory-db\\.git\\logs\\refs\\remotes\\origin\\advertise-ghsa" for appending
  -> Signature name or email must not contain '<', '>' or \n
```

More details:
```
C:\Users\user\Desktop\rust_projects\project>cargo audit --version
cargo-audit-audit 0.20.0

C:\Users\user\Desktop\rust_projects\project>cargo --version
cargo 1.79.0 (ffa9cf99a 2024-06-03)

C:\Users\user\Desktop\rust_projects\project>rustc --version
rustc 1.79.0 (129f3b996 2024-06-10)

C:\Users\user\Desktop\rust_projects\project>cd C:\Users\user\.cargo\advisory-db
The system cannot find the path specified.
```

Running Windows 10 Pro 22H2
Result is the same using both CMD and PowerShell.

**Comment by MeGaGiGaGon** [comment]
That's one of the things confusing me, as far as I can tell `C:\Users\user\.cargo\advisory-db` doesn't exist.
![image](https://github.com/user-attachments/assets/a13db401-8bcb-4358-bbd3-231be628fffc)

**Comment by y-mx-b** [comment]
I had this same error, so I checked my config with `git config --list` just in case, because of that top-level `-> Signature name or email must not contain '<', '>' or \n` error. It turned out that in my `.gitconfig`, my email was surrounded by angle brackets for some reason. Fixed by changing my email so that it was delineated with quotation marks instead. @MeGaGiGaGon perhaps this might work for you?

The error message being so cryptic, especially without adding `display_error_chain`, is rather unfortunate, but this doesn't seem to be on the rustsec team?

**Comment by MeGaGiGaGon** [comment]
@y-mx-b Thanks, you may be correct. I say may, because when I checked `git config --list` my email was indeed `user.email=<107241144+MeGaGiGaGon@users.noreply.github.com>` After I removed the braces and re-installed `cargo audit`, it worked as expected. But even when I reverted my email back to the version with `<>`, it still works. Maybe the underlying issue was fixed at some point? Since even with the `<>`, everything else still worked fine. I also found #1067, not sure if that actually fixed the error message though. Since a likely fix has been found for anyone else coming across this, and I can't reproduce the issue on a fresh install, and #1029 already exists with the commentary on improving the failure error messages, I'm closing this.

**Comment by y-mx-b** [comment]
> @y-mx-b Thanks, you may be correct. I say may, because when I checked `git config --list` my email was indeed `user.email=<107241144+MeGaGiGaGon@users.noreply.github.com>` After I removed the braces and re-installed `cargo audit`, it worked as expected. But even when I reverted my email back to the version with `<>`, it still works. Maybe the underlying issue was fixed at some point? Since even with the `<>`, everything else still worked fine. I also found #1067, not sure if that actually fixed the error message though. Since a likely fix has been found for anyone else coming across this, and I can't reproduce the issue on a fresh install, and #1029 already exists with the commentary on improving the failure error messages, I'm closing this.

The reason everything works fine after successfully running `cargo audit` for the first time is because it only errors out on the initial clone of `advisory-db`. I initially found this out because I attempted to manually clone the database into `.cargo/advisory-db` and subsequent usage of `cargo audit` was fine.

**Comment by MeGaGiGaGon** [comment]
Yep, you are once again right. After deleting `C:\Users\user\.cargo\advisory-db` the problem came back. I wonder if it's something like VSC/VS/PyCharm that I used to set that email that game it the `<>`s. Regardless, I also agree that this doesn't look like a `rustsec` issue and is instead an issue with one of those tools/a dependent, and the bad error message handling is already being worked on, so I don't see a reason to keep this open.
